entrypoint.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+composer install \
```

Comment:
This is a great improvement!

I would like to mention that perhaps we shouldn't run `composer install` and `npm install` in the entrypoint as every time that you boot the container it'll attempt to download dependencies and that's very slow.
Also, you have already added that step in the `Dockerfile`, so this becomes redundant.

Reply:
My initial attempt at commit 82023f8 was doing that, but when trying to run it from a fresh environment the mix compile step would sometimes fail, and the suggested solution at StackOverflow was to run `npm install` again right before attempting `npm run prod`. Skipping on `composer install` would similarly cause issues during the `php artisan migrate` step, but I too would like to explore further into safely removing those steps from the entrypoint.

Reply:
Alright, those redundant steps are gonna and it seems to be for good this time.

Turns out mounting the local directory inside the container overwrote the files added by the install, so the `docker-compose.yml` had to be edited not to mount all that but only the folder with the dictionaries, and in my shallow testing both the feature of adding dictionaries and the database persistence weren't broken by this.

Having the entrypoint reduced so much also introduced a race condition where the PHP container could start before the MySQL one was ready to accept connections, so I had to add a healthcheck for that, also in the `docker-compose.yml`.

@simjanos-dev I would say this is functional right now, except maybe setting `COMPOSER_ALLOW_SUPERUSER`in the `.env` file to keep things tidy and bringing `docker-compose-dev.yml` and `docker-compose-jellyfin.yml` to parity. Let me know your thoughts.

Reply:
I have noticed this also makes the `sudo chmod 777 ./* -R` step unnecessary since the problematic files have now their permissions set in their respective `Dockerfile`. However, this might have broken logs, should be taken into account before merging.

Reply:
@sergiolaverde0 I've seen this, this is great, I will test it and merge it. I'll just need some time to figure out how to do things with github. I've only used it for personal development before, it might take some time.

Reply:
I tested the code, and it works great. I'm sorry I messed up your pull request with V0002. I never done pull requests before, and I've just learned how to handle them.

Can you please make it up to date with V0002 and modify the update guide as well? 

I think if we use this dockerfile and yml then if someone pulls the codes and updates to a newer version, they will need to run "docker compose build --no-cache" so the composer and npm commands run again.

Do I understand it correctly, that it will only merge into dev, and not main if I click on the "Merge pull request" button? I would like to keep the main branch for full version updates.

Reply:
But I'm not sure how can I update it if there's a new version of LinguaCafe. Can you please modify the readme update guide on how to update your container if there is a new version? I'll merge it after.

(Sorry for the messy comments, I'm still learning GitHub,)

Reply:
you build a new image with the updated config instead of the current workflow which overwrites de old one (by installing new packages in the mounted location). I just built the images by running these two commands in the repo's root directory:
```
docker build . --file docker/PhpDockerfile --tag linguacafe-webserver:0.0.3
docker build . --file docker/PythonDockerfile --tag linguacafe-python-service:0.0.3
```
Where `0.0.3` is the version I have arbitrarily given to them. Creating a new version would then require to just run the same commands with the repository in a new state and change the version tag. This even allows to run older versions of the container if needed to hunt for bugs or regressions.

Reply:
For a shorter solution while the process of building and distributing images is setup, doing:
```
docker compose down
docker image rm linguacafe-webserver linguacafe-python-service
docker compose up -d
```
Should work just fine, and is what I will go with in the `README.md`for the time being, I will edit it in a couple of minutes. Then I will try to look into the Github Container Registry at some moment of the week so that the images are built once here and users simply download them.

(It's fine, I'm new to GitHub too and this is my first pull request)

Reply:
I think I understand, but could you please modify the readme file install and update section using "v0002" as a tag, so I won't mess up? Haven't used tags yet in docker.

Reply:
With my latest commit, it should be ready for merge. Let me know if you encounter any issue that you suspect was introduced by it.

Reply:
Thank you! It looks great. I upload my git Issues, test it myself and merge it in today.